src/easy_config/contrib/click.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+
+"""A wrapper for generating options for a :mod:`click` command from an :class:`easy_config.EasyConfig`."""
+
+import dataclasses
+from typing import Callable, Type
+
+import click
+
+from easy_config import EasyConfig
+
+__all__ = [
+    'args_from_config',
+]
+
+# A callable that can take in anything, but gives back nothing
+GenericCallable = Callable[..., None]
```

Comment:
Found the "right way": https://mypy.readthedocs.io/en/latest/generics.html#declaring-decorators

And click commands can return now so it should be `[..., Any]` instead of `None`: https://click.palletsprojects.com/en/7.x/commands/#command-return-values